server/src/services/storage.service.ts

```diff
@@ -12,9 +19,32 @@ const docsMessage = `Please see https://immich.app/docs/administration/system-in
 
 @Injectable()
 export class StorageService extends BaseService {
-  @OnEvent({ name: 'AppBootstrap' })
-  async onBootstrap() {
+  private detectMediaLocation(): string {
     const envData = this.configRepository.getEnv();
+    if (envData.storage.mediaLocation) {
+      return envData.storage.mediaLocation;
+    }
+
+    const targets: string[] = [];
+    const candidates = ['/data', '/usr/src/app/upload'];
+
+    for (const candidate of candidates) {
+      const exists = this.storageRepository.existsSync(candidate);
+      if (exists) {
+        targets.push(candidate);
+      }
+    }
+
+    if (targets.length === 1) {
+      return targets[0];
+    }
+
+    return '/usr/src/app/upload';
```

Comment:
I think this should be `/data`? If both `/data` and `/usr/src/app/upload` exist for whatever reason, surely we'd want to use the newer one.

Reply:
So it's less about both existing and more about the likelihood of the legacy one existing at all. Plus, in every situation this keeps the old behavior. Another way to think about it is only, specifically, in the scenario where /data is there and /usr/src/app/upload is NOT do we assume the new /data default. Everything else keeps the current, old behavior.